app/models/user.rb

```diff
@@ -505,6 +505,17 @@ def latest_location
     recent_locations.last
   end
 
+  def self.recently_active_users(limit = 5, order = 'last_updated DESC')
+    Rails.cache.fetch('users/active', expires_in: 24.hours) do
+      User.select('rusers.username, rusers.id, MAX(node_revisions.timestamp) AS last_updated')
+        .joins("INNER JOIN `node_revisions` ON `node_revisions`.`uid` = `rusers`.`id` ")
+        .where("node_revisions.status = 1")
```

Comment:
I wonder if it would make things more efficient to add `.where("rusers.status = 1") as well here, and to the `.select()` above? We do have tens of thousands of user records. Just an idea, not sure if the join already makes this efficient!

Reply:
@jywarren I'll add that and see if it works.